src/controllers/faculty_controller.py

```diff
@@ -0,0 +1,70 @@
+from flask import Blueprint, request, jsonify
+from pydantic import ValidationError
+
+from src.services.faculty_service import FacultyService
+from src.dtos.faculty_dto import FacultyDTO
+from src.models.faculty import Faculty
+
+faculty_service = FacultyService()
+faculty_blueprint = Blueprint('faculty', __name__)
+
+
+@faculty_blueprint.route('/faculty', methods=['GET'])
+def get_all_faculties():
+    faculties = faculty_service.get_all_faculties()
+    return jsonify(faculties), 200
+
+
+@faculty_blueprint.route('/faculty/<int:faculty_id>', methods=['GET'])
+def get_faculty(faculty_id):
+    faculty = faculty_service.get_faculty_by_id(faculty_id)
+    if faculty:
+        return jsonify(faculty), 200
+    return jsonify({"error": "Faculty not found"}), 404
+
+
+@faculty_blueprint.route('/faculty', methods=['POST'])
+def create_faculty():
+    # Grab the raw JSON
+    raw_json = request.get_json()
+
+    # Validate against the Pydantic DTO
+    try:
+        dto = FacultyDTO(**raw_json)
+    except ValidationError as e:
+        return jsonify({"error": str(e)}), 400
+
+    # Convert the DTO to a Faculty model
+    faculty_obj = dto.to_model()
```

Comment:
Inside Model Faculty you should have constructor that is accepting DTOModel and creating Model. Remove this to_model method.